### PR TITLE
New promo card Supercapacitors

### PR DESCRIPTION
**Pull request by kberg**
image

### CODE REVIEW

File: src/server/Player.ts [hunk 1]
```diff
@@ -740,13 +741,23 @@ export class Player {
...
+    } else {
+      this.heat += this.energy;
+      this.energy = 0;
+      this.finishProductionPhase();
+    }
+  }
+
+  public finishProductionPhase() {
     this.megaCredits += this.production.megacredits + this.terraformRating;
-    this.heat += this.energy;
```
Comment (kberg) [inline src/server/Player.ts:744]: This is just reordering.

File: src/server/Player.ts [hunk 2]
```diff
@@ -740,13 +741,23 @@ export class Player {
...
-    this.heat += this.production.heat;
-    this.energy = this.production.energy;
```
Comment (kberg) [inline src/server/Player.ts:746]: But this is replaced with this.energy +=, below.

**Review by d-little — APPROVED** [review]
Fast work, great job!

File: src/server/cards/promo/Supercapacitors.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+          b.br;
+          b.production((pb) => pb.megacredits(1));
+        }),
+        description: 'Increase M€ production 1 step.',
+      },
+    });
+  }
+
+  public static onProduction(player: Player) {
+    if (player.energy === 0) {
```
Comment (nwai90) [inline src/server/cards/promo/Supercapacitors.ts:35]: This is incorrect IMO. If a player happens to have positive energy production but 0 energy at the end of the round (i.e. they spent all their energy during the generation),  they will never produce any resources as `player.finishProductionPhase()` will never be called

**Review by nwai90 — COMMENTED** [review]
It seems there is a bug in this implementation

**Pull request merged**